applications/RomApplication/python_scripts/rom_manager.py

```diff
@@ -569,25 +613,65 @@ def __LaunchRunHROM(self, mu_run, use_full_model_part):
             simulation.Run()
             self.QoI_Run_HROM.append(simulation.GetFinalData())
 
-    def __LaunchTrainNeuralNetwork(self):
+    def _LaunchTrainNeuralNetwork(self, mu_train, mu_validation):
         if not have_tensorflow:
             err_msg = f'Tensorflow module not found. Please install Tensorflow in to use the "ann_enhanced" option.'
             raise Exception(err_msg)
-        
-        rom_nn_trainer = RomNeuralNetworkTrainer(self.general_rom_manager_parameters)
+
+        rom_nn_trainer = RomNeuralNetworkTrainer(self.general_rom_manager_parameters, mu_train, mu_validation, self.data_base)
         model_name = rom_nn_trainer.TrainNetwork()
+        self.data_base.add_to_database("Neural_Network", mu_train , None)
         rom_nn_trainer.EvaluateNetwork(model_name)
-        
 
-    def __LaunchTestNeuralNetworkReconstruction(self):
+
+    def _LaunchTestNeuralNetworkReconstruction(self):
         if not have_tensorflow:
             err_msg = f'Tensorflow module not found. Please install Tensorflow in to use the "ann_enhanced" option.'
             raise Exception(err_msg)
-        
+
         rom_nn_trainer = RomNeuralNetworkTrainer(self.general_rom_manager_parameters)
         model_name=self.general_rom_manager_parameters["ROM"]["ann_enhanced_settings"]["online"]["model_name"].GetString()
```

Comment:
I think this system for testing the NN won't work very well anymore, as the networks are saved by their hash now. Maybe we could integrate this testing routine into the general Test() method and just evaluate the network that coincides with the given parameters, if it exists.

Reply:
It was changed, however to know the name of the model one needs both the mu_train and mu_test. 

Currently this method is not called in the RomManger, when doing Fit or Test. Where would it be called?